src/test/java/com/jwplayer/southpaw/util/ByteArraySetTest.java

```diff
@@ -46,6 +74,651 @@ public ByteArraySet createSmallSet() {
+
+        List<ByteArray> vals = new ArrayList<>();
+        for (ByteArray val: originalVals) {
+            vals.add(val);
+        }
+        Collections.shuffle(vals);
+
+        int forceMergerSize = (new Random()).nextInt(size);
+        List<String> insertedStringVals = new ArrayList<>();
+        for (ByteArray val: vals) {
+            insertedStringVals.add(new String(val.getBytes()));
+            assertTrue(set.add(val));
+            if (set.size() == forceMergerSize) {
+                // Forces the merger of the fronting set into the list of chunks
+                set.serialize();
+            }
+        }
+
+        int deserializedCheckSize = (new Random()).nextInt(size);
+        int currentSize = size;
+        List<String> deletedStringVals = new ArrayList<>();
+        for (ByteArray val: originalVals) {
+            if (currentSize == deserializedCheckSize) {
+                if (currentSize != ByteArraySet.deserialize(set.serialize()).size()) {
+                    assertEquals("",
+                            String.format("\nInserted Vals: %s\nDeleted Vals: %s\nMerger Size: %d\nCheck Size: %d\n",
+                                    insertedStringVals.toString(),
+                                    deletedStringVals.toString(),
+                                    forceMergerSize,
+                                    deserializedCheckSize));
+                }
+            }
+            assertTrue(set.remove(val));
+            assertFalse(set.contains(val));
+            deletedStringVals.add(new String(val.getBytes()));
+            currentSize -= 1;
+            assertEquals(currentSize, set.size());
+        }
+    }
+
+    @Test
+    public void reallySmallRandomSerializeDerializeSizeCheckRemove() {
+        testRandomSerializeDerializeSizeCheckRemove(SIZE_REALLY_SMALL);
+    }
+
+    @Test
+    public void smallRandomSerializeDerializeSizeCheckRemove() {
+        testRandomSerializeDerializeSizeCheckRemove(SIZE_SMALL);
+    }
+
+    @Test
+    public void regularRandomSerializedRemove() {
+        testRandomSerializeDerializeSizeCheckRemove(SIZE_REGULAR);
+    }
+
+    @Test
+    public void bigRandomSerializeDerializeSizeCheckRemove() {
+        testRandomSerializeDerializeSizeCheckRemove(SIZE_BIG);
+    }
+
+    @Test
+    public void reallyBigRandomSerializeDerializeSizeCheckRemove() {
+        testRandomSerializeDerializeSizeCheckRemove(SIZE_REALLY_BIG);
+    }
+
+    private void testSimilarRemove(boolean forceMerger) {
+        System.out.println("Testing remove method for similar inputs with " + forceMerger + " forceMerger");
+        ByteArraySet set = new ByteArraySet();
+
+        List<ByteArray> vals = new ArrayList<>();
+        vals.add(new ByteArray("bytear"));
+        vals.add(new ByteArray("ByteAr"));
+        vals.add(new ByteArray("bytea0"));
+        vals.add(new ByteArray("0ytear"));
+
+        for (ByteArray val: vals) {
+            assertTrue(set.add(val));
+        }
```

Comment:
```suggestion
        List<ByteArray> vals = new ArrayList<>();
        vals.add(new ByteArray("bytear"));
        vals.add(new ByteArray("ByteAr"));
        vals.add(new ByteArray("bytea0"));
        vals.add(new ByteArray("0ytear"));

        ByteArraySet set = new ByteArraySet(vals);
```
Can be simplified with

Reply:
ByteArraySet cannot be instantiated with a List, but simplified with addAll